### PR TITLE
LUCENE-8640: added changes for the validation of valid dateString

### PR DESCRIPTION
**Pull request by MighTguY**

### CODE REVIEW

File: solr/core/src/test/org/apache/solr/schema/DateRangeFieldTest.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
 import org.junit.BeforeClass;
 import org.junit.Test;
 
-public class DateRangeFieldTest extends SolrTestCaseJ4 {
+public class  DateRangeFieldTest extends SolrTestCaseJ4 {
```
Comment (gd-spb-e1m) [inline solr/core/src/test/org/apache/solr/schema/DateRangeFieldTest.java]: pls exclude from PR

File: lucene/spatial-extras/src/java/org/apache/lucene/spatial/prefix/tree/DateRangePrefixTree.java [hunk 2]
```diff
@@ -500,4 +516,18 @@ public Calendar parseCalendar(String str) throws ParseException {
...
+  private  void isValidDateDelimeter(String str, int offset, char delim) {
+    if(str.charAt(offset) != delim) {
+      throw new IllegalArgumentException("Not the valid delimeter for Position"+offset);
+    }
+  }
+
+  private int parseIntegerAndValidate(String  str, int offset, int min, int max) {
+    int val = Integer.parseInt(str.substring(offset, offset+2));
+    if((val < min ) || (val>max)) {
+      throw new IllegalArgumentException("Not valid date.");
```
Comment (gd-spb-e1m) [inline lucene/spatial-extras/src/java/org/apache/lucene/spatial/prefix/tree/DateRangePrefixTree.java]: I suppose expected and actual character might be reported here, assuming enclosing ParseException report whole input string and the position.  The same is applicable to IAE above.

File: solr/core/src/test/org/apache/solr/schema/DateRangeFieldTest.java (hunk 1 above)
Comment (MighTguY) [reply solr/core/src/test/org/apache/solr/schema/DateRangeFieldTest.java]: done

File: lucene/spatial-extras/src/java/org/apache/lucene/spatial/prefix/tree/DateRangePrefixTree.java (hunk 2 above)
Comment (MighTguY) [reply lucene/spatial-extras/src/java/org/apache/lucene/spatial/prefix/tree/DateRangePrefixTree.java]: I have incorporated the same.

Comment (MighTguY) [reply lucene/spatial-extras/src/java/org/apache/lucene/spatial/prefix/tree/DateRangePrefixTree.java]: @gd-spb-e1m I have incorporated the same, please review

**Comment by munendrasn** [comment]
Closing this as LUCENE-8640 is resolved

**Pull request closed without merging**